src/utils/utils.jl

```diff
@@ -115,57 +116,77 @@ function validate_exported_names(mod::Module)
     return is_valid
 end
 
+# Make match_fn optional
+compare_values(x, y; kwargs...) = compare_values(nothing, x, y; kwargs...)
+
+# Get the default match_fn if necessary. Only call when you know you're done recursing
+_fetch_match_fn(match_fn::Function) = match_fn
```

Comment:
This is quite a complicated workaround... I understand it avoids code repetition but it makes the code harder to read

Reply:
Would
```julia
(isnothing(match_fn) ? isequivalent : match_fn)(a, b)
```
be more readable than
```julia
_fetch_match_fn(match_fn)(a, b)
```
?